tests/perf-system/benchmarks/benchmarks.py

```diff
@@ -0,0 +1,513 @@
+        str(sig_ms_interval),
+        "--worker-threads",
+        str(worker_threads),
+        "--workspace",
+        ccf_dir / "workspace",
+    ]
+
+    for address in ccf_node_ips:
+        sandbox_n_args.append("-n")
+        sandbox_n_args.append("ssh://" + address)
+
+    if ccf_app == "js_logging":
+        sandbox_n_args += ["--js-app-bundle", "../samples/apps/logging/js/"]
+    else:
+        sandbox_n_args += ["-p", "samples/apps/logging/liblogging"]
+
+    if enclave_type == "sgx":
+        sandbox_n_args += ["-t", "sgx", "-e", "release"]
+
+    if enclave_type == "sgx":
+        ccf_build_dir = ccf_dir / "build-sgx"
+    else:
+        ccf_build_dir = ccf_dir / "build-virtual"
+
+    ccf_server_process = subprocess.Popen(sandbox_n_args, cwd=ccf_build_dir)
+    time.sleep(120)
+
+    LOG.info("Preparing workloads using piccolo generator")
+
+    # TOOD: remove this hack
+    generator_dir = ccf_dir / "tests" / "perf-system" / "generator"
+    sys.path.insert(0, str(generator_dir))
+    import generator
+
+    for workload_number, write_percentage in enumerate(write_percentages):
+        LOG.info(
+            f"Generating workload {workload_number} with {write_percentage}% writes"
+        )
+
+        for piccolo_number in range(number_of_piccolos):
+            writes = random.sample(
+                range(number_of_requests),
+                int(number_of_requests * write_percentage / 100),
+            )
+            LOG.info(
+                f"In workload {workload_number} piccolo {piccolo_number} will write {len(writes)} messages and read {number_of_requests - len(writes)} messages"
+            )
+            workload = generator.Messages()
+            for i in range(number_of_requests):
+                if i in writes:
+                    body = {
+                        "id": i,
+                        "msg": str(
+                            "".join(
+                                random.choices(
+                                    string.ascii_letters, k=logging_message_length
+                                )
+                            )
+                        ),
+                    }
+                    workload.append(
+                        "/log/public",
+                        "POST",
+                        body=json.dumps(body),
+                    )
+                else:
+                    workload.append(
+                        f"/log/public?id={i}",
+                        "GET",
+                    )
+            workload.to_parquet_file(
+                experiment_dir
+                / f"writes{write_percentage}%"
+                / f"piccolo{piccolo_number}"
+                / "input.parquet"
+            )
+    LOG.info("Workloads generated")
+
+    LOG.info("Checking CCF service is live & new before starting experiment")
+    addresses_of_ccf_nodes = [f"https://{i}" for i in ccf_node_ips]
+    for id, address in enumerate(addresses_of_ccf_nodes):
```

Comment:
The Network (https://github.com/microsoft/CCF/blob/9e0e46c41a306d10dc128f3f3672ad16a58cc089/tests/infra/network.py) and CCFRemote classes take care of file copy, node startup, waiting for nodes to be ready etc. It would be preferable not to duplicate this logic.